clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocol.java

```diff
@@ -102,53 +106,97 @@
         new Field(PARTITIONS_KEY_NAME, new ArrayOf(Type.INT32)));
 
     public static final Schema SUBSCRIPTION_V0 = new Schema(
-            new Field(TOPICS_KEY_NAME, new ArrayOf(Type.STRING)),
-            new Field(USER_DATA_KEY_NAME, Type.NULLABLE_BYTES));
+        new Field(USER_DATA_KEY_NAME, Type.NULLABLE_BYTES),
```

Comment:
Hmm.. reordering fields from bytes would be breaking changes, is it intentional?

Reply:
More generally speaking, we should only add new fields in new versions but should not remove / reorder existing fields.

Reply:
Note the `struct.set` calls does not need to be in order, as the schema definition's ordering is the source of truth.

Reply:
Sorry yeah should not have been reordered